core/rootchain.go

```diff
@@ -0,0 +1,127 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	// "github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/icstglobal/go-icst/chain"
+	"github.com/icstglobal/go-icst/chain/eth"
+	"github.com/icstglobal/plasma/core/types"
+
+	"strings"
+	"time"
+)
+
+const (
+	DepositEventName     = "Deposit"
+	ExitedStartEventName = "ExistedStart"
+)
+
+type RootChain struct {
+	chain    chain.Chain
+	sub      map[string]func(eventName string, log ethtypes.Log) // map topic0 to name
+	cxAbi    abi.ABI
+	operator *Operator
+}
+
+type DepositEvent struct {
+	Depositor    common.Address
+	DepositBlock *big.Int
+	Token        common.Address
+	Amount       *big.Int
+	Raw          ethtypes.Log // Blockchain specific contextual infos
+}
+
+// chain
+func NewRootChain(url string, abiStr string, operator *Operator) (*RootChain, error) {
+
+	client, err := ethclient.Dial(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect eth rpc endpoint {%v}, err is:%v", url, err)
+	}
+	blc := eth.NewChainEthereum(client)
+	chain.Set(chain.Eth, blc)
```

Comment:
why not read chain type from config?

if `rootchain` is really `Ethereum` specific, should we remove chain type config?

Reply:
the type has been removed.